src/Apropos/Gen/Api.hs

```diff
@@ -0,0 +1,49 @@
+{- |
+Module: Apropos.Gen.Api
+Description: Misc. Plutus generators.
+Maintainer: jack@mlabs.city
+
+`Gen`s for `Plutus.V1.Ledger.Api` types.
+-}
+module Apropos.Gen.Api (
+  builtinByteString,
+  builtinData,
+) where
+
+import Apropos.Gen (Gen, element, int, linear, list, singleton)
+import Apropos.Gen.Extra (integer, pair, sha256)
+import PlutusTx (Data (B, Constr, I, List, Map))
+import PlutusTx.Builtins.Internal (
+  BuiltinByteString (BuiltinByteString),
+  BuiltinData (BuiltinData),
+ )
+
+-- | `Gen` for Plutus `BuiltinByteString`.
+builtinByteString :: Gen BuiltinByteString
+builtinByteString = do
+  bs <- sha256
+  return $ BuiltinByteString bs
+
+-- | `Gen` for Plutus `BuiltinData`.
+builtinData :: Gen BuiltinData
+builtinData = do
+  d <- data'
+  return $ BuiltinData d
+
+-- | `Gen` for Plutus `Data`.
+data' :: Gen Data
+data' = do
+  let mapGen :: Gen [(Data, Data)]
+      mapGen = list (linear 0 20) $ pair data' data'
+  n <- integer $ linear minBound maxBound
```

Comment:
I think generating these upfront and using element is worse for performance than using choice and generating them in their list entries.

Reply:
Interesting! ELI5?

Reply:
The same would apply for sum types, like [`scriptPurpose`](https://github.com/mlabs-haskell/apropos-tx/pull/41/files#diff-3bdee9ceafcb87ab1d7e8d77038ba042c553ab4204647170acbe115fcbf85a83R107-R108).

Reply:
The same would apply for sum types, like [`scriptPurpose`](https://github.com/mlabs-haskell/apropos-tx/pull/41/files#diff-3bdee9ceafcb87ab1d7e8d77038ba042c553ab4204647170acbe115fcbf85a83R107-R108).